Utilities/sysinfo.cpp

```diff
@@ -51,6 +50,20 @@ bool utils::has_xop()
 	return g_value;
 }
 
+bool utils::is_haswell()
+{
+	static const int g_model = ((get_cpuid(1, 0)[0] >> 4) & 0xf) + (((get_cpuid(1, 0)[0] >> 16) & 0xf) << 4);
```

Comment:
the assignment for g_model is the same for both haswell & broadwell cases. Refactor into seperate method, eg utils::get_cpu_model()

Reply:
Restructured it.